### PR TITLE
Cancel product - rebased - WIP

### PR DESCRIPTION
**Pull request by matks**
⚠️ STILL WIP, I check the rebase did not break something

REBASE OF https://github.com/PrestaShop/PrestaShop/pull/17149

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Cancel product feature on migrated order detail page.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | Does it break backward compatibility? yes/no
| Deprecations? | Does it deprecate an existing feature? yes/no
| Fixed ticket? | #16650
| How to test?  | Make an order in frontend, then go the order page in BO, click on "cancel product" at the top of the page, then select one or more product and save. New Order page URL:  **/admin-dev/sell/orders/orders/8/view** . (where 8 is the order ID)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17458)

### CODE REVIEW

File: src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php [hunk 1]
```diff
@@ -368,8 +368,8 @@ private function getOrderProducts(Order $order): OrderProductsForViewing
             $product['quantity_refundable'] = $product['product_quantity'] - $product['product_quantity_return'] - $product['product_quantity_refunded'];
             $product['amount_refundable'] = $product['total_price_tax_excl'] - $product['total_refunded_tax_excl'];
             $product['amount_refundable_tax_incl'] = $product['total_price_tax_incl'] - $product['total_refunded_tax_incl'];
-            $product['displayed_max_refundable'] = $taxCalculationMethod === PS_TAX_EXC ? $product['amount_refundable'] : $product['amount_refundable_tax_incl'];
-            $resumeAmountKey = $taxCalculationMethod === PS_TAX_EXC ? 'total_refunded_tax_excl' : 'total_refunded_tax_incl';
+            $product['displayed_max_refundable'] = $order->getTaxCalculationMethod() ? $product['amount_refundable'] : $product['amount_refundable_tax_incl'];
```
Comment (matks) [inline src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php:371]: Not sure if this change is right
@jolelievre can you tell me ?

File: src/PrestaShopBundle/Form/Admin/Sell/Order/CancelProductType.php [hunk 2]
```diff
@@ -147,10 +157,11 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
             ])
             ->add('save', SubmitType::class, [
                 'attr' => [
-                    'class' => 'cancel-product-element save btn btn-primary ml-3',
+                    'class' => 'partial-refund cancel-product-element save btn btn-primary ml-3',
```
Comment (matks) [inline src/PrestaShopBundle/Form/Admin/Sell/Order/CancelProductType.php:160]: Not sure if it's right

File: src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php (hunk 1 above)
Comment (jolelievre) [reply src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php:371]: Nope this one I fixed there was a bug, the correct code is the one from 177x

**Comment by matks** [comment]
@matthieu-rolland I guess I can close this PR now ?

**Comment by jolelievre** [comment]
I guess so!

**Comment by jolelievre** [comment]
Fixed by https://github.com/PrestaShop/PrestaShop/pull/17149

**Pull request closed without merging**